cmd2/argparse_custom.py

```diff
@@ -724,21 +724,70 @@ def _format_args(self, action, default_metavar) -> str:
         return result
 
 
+class _UnloadableSubParsersAction(argparse._SubParsersAction):
+    """Extends the argparse internal SubParsers action to allow sub-parsers to be removed dynamically"""
+    def remove_parser(self, name):
+        """Removes a sub-parser from the sub-parsers group"""
+        for choice_action in self._choices_actions:
+            if choice_action.dest == name:
+                self._choices_actions.remove(choice_action)
+                break
+
+        subparser = self._name_parser_map[name]
+        to_remove = []
+        for name, parser in self._name_parser_map.items():
+            if parser is subparser:
+                to_remove.append(name)
+        for name in to_remove:
+            del self._name_parser_map[name]
+
+
 # noinspection PyCompatibility
 class Cmd2ArgumentParser(argparse.ArgumentParser):
     """Custom ArgumentParser class that improves error and help output"""
 
-    def __init__(self, *args, **kwargs) -> None:
-        if 'formatter_class' not in kwargs:
-            kwargs['formatter_class'] = Cmd2HelpFormatter
-
-        super().__init__(*args, **kwargs)
+    def __init__(self,
```

Comment:
1) It's really annoying to use because all of the IDE help is lost with *args/**kwargs. Also, this is something that @tleonhardt identified as an antipattern on one of the previous iterations.
2) The way python default values work, a list declared as a default value is shared across all instantiations of an object. If it is subsequently modified it is modified for all instances. Generally considered unsafe.

Reply:
I really like having good help in PyCharm or VSCode.